pkgs/by-name/mq/mqttx-cli/package.nix

```
@@ -0,0 +1,74 @@
+{
+  fetchFromGitHub,
+  fetchYarnDeps,
+  fetchpatch,
+  lib,
+  mqttx-cli,
+  nodejs,
+  stdenv,
+  testers,
+  yarnConfigHook,
+  yarnInstallHook,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "mqttx-cli";
+  version = "1.11.0";
+
+  src = fetchFromGitHub {
+    owner = "emqx";
+    repo = "MQTTX";
+    rev = "refs/tags/v${finalAttrs.version}";
+    hash = "sha256-IPIiSav6MPJmzetBgVw9fLGPjJ+JKS3oWMEfCJmEY84=";
+  };
+
+  patches = [
+    # moves @faker-js/faker from devDependencies to dependencies
+    # because the final package depends on it
+    # https://github.com/emqx/MQTTX/pull/1801
+    (fetchpatch {
+      url = "https://github.com/emqx/MQTTX/commit/3d89c3a08477e9e2b5d83f2a222ceaa8c08e50ce.patch";
+      hash = "sha256-Rd6YpGHsvAYD7/XCJq6dgvGeKfOiLh7IUQFr/AQz0mY=";
+    })
+  ];
+
+  yarnOfflineCache = fetchYarnDeps {
+    yarnLock = "${finalAttrs.src}/cli/yarn.lock";
+    hash = "sha256-vwPwSE6adxM1gkdsJBq3LH2eXze9yXADvnM90LsKjjo=";
+  };
+
+  nativeBuildInputs = [
+    nodejs
+    yarnConfigHook
+    yarnInstallHook
+  ];
+
+  preConfigure = ''
+    cd cli
```

```suggestion
    export HOME="$(mktemp -d)"
    cd cli
```
Might fix the build failure.

---

I am not able to reproduce the build failure on my x86_64-darwin machine...
I know that hydra is using rosetta on aarch64-darwin to build x86_64-darwin packages, so it may be an issue with that. I will try building it on an aarch64-darwin machine when I have the time.

---

Did you rebase the PR on master before building? (that is almost what ofborg does)

---

I ran nixpkgs-review on a x86_64-darwin machine.

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 337730`

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>mqttx-cli</li>
  </ul>
</details>